Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/DeleteCaseTask.java

```diff
@@ -0,0 +1,850 @@
+                    logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()));
+                    return;
+                }
+
+                progress.progress(Bundle.DeleteCaseTask_progress_gettingManifestPaths());
+                logger.log(Level.INFO, String.format("Getting manifest file paths for %s", caseNodeData.getDisplayName()));
+                try {
+                    getManifestFilePaths();
+                } catch (IOException | CoordinationServiceException ex) {
+                    logger.log(Level.SEVERE, String.format("An error occurred getting the manifest file paths", caseNodeData.getDisplayName()), ex);
+                    return;
+                } catch (InterruptedException ex) {
+                    logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()), ex);
+                    return;
+                }
+                logger.log(Level.INFO, String.format("Found %d manifest file path(s) for %s", manifestFilePaths.size(), caseNodeData.getDisplayName()));
+
+                if (Thread.currentThread().isInterrupted()) {
+                    logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()));
+                    return;
+                }
+
+                if (!manifestFilePaths.isEmpty()) {
+                    /*
+                     * Acquire exclusive locks for the auto ingest job manifest
+                     * files for the case, if any. Manifest file locks are
+                     * acquired by the auto ingest node (AIN) input directory
+                     * scanning tasks when they look for auto ingest jobs to
+                     * enqueue, and by the AIN job execution tasks when they do
+                     * a job. Acquiring these locks here ensures that the
+                     * scanning tasks and job execution tasks cannot do anything
+                     * with the auto ingest jobs for a case during case
+                     * deletion.
+                     */
+                    progress.progress(Bundle.DeleteCaseTask_progress_acquiringManifestLocks());
+                    logger.log(Level.INFO, String.format("Acquiring exclusive manifest file locks for %s", caseNodeData.getDisplayName()));
+                    try {
+                        if (!acquireManifestFileLocks()) {
+                            logger.log(Level.INFO, String.format("Could not delete %s because at least one manifest file lock was already held by another host", caseNodeData.getDisplayName()));
+                            return;
+                        }
+                    } catch (IOException | CoordinationServiceException ex) {
+                        logger.log(Level.SEVERE, String.format("Could not delete %s because an error occurred acquiring the manifest file locks", caseNodeData.getDisplayName()), ex);
```

Comment:
Do the manifest locks need to be released here?

Reply:
See discussion in Jira.